bykov_ia/task2/Source.cpp

```diff
@@ -0,0 +1,113 @@
+#include <stdio.h>
+#include <iostream>
+using std::cout;
+using std::endl;
+
+class Polynomial {
+	private:
+		int power;
+		float *coefficients;
+	public:
+		Polynomial(int _power, float *_coefficients) {
+			try {
+				if (_power < 0 || _power > 12)
+					throw "IncorrectPower";
+				power = _power;
+				coefficients = new float[power + 1];
+				for (int i = 0; i <= power; i++)
+					coefficients[i] = _coefficients[i];
+			}
+			catch(char *s) {
+				printf("%s\n", s);
+			}
+		}
+
+		~Polynomial() {
+			delete[] coefficients;
+		}
+
+		int getPower() {
+			return power;
+		}
+
+		float getCoefficient(int num) {
+			try {
+				if (num < 0 || num > power)
+					throw "IncorrectNumberCoefficient";
+				return coefficients[num];
+			}
+			catch(char* s) {
+				printf("%s\n", s);
+				return -1;
+			}
```

Comment:
Так неверно. -1 не является выделенным значением. Раз решили проверять номер, то нужно просто бросить исключение. Например, так:
```suggestion
	if (num < 0 || num > power)
		throw "IncorrectNumberCoefficient";
	return coefficients[num];
```